webapp/jira.py

```diff
@@ -59,7 +61,10 @@ def __request__(self, method: str, url: str, data: dict = {}, params: dict = {})
             try:
                 return response.json()
             except json.JSONDecodeError:
-                return response.status_code
+                return {
+                    "status_code": response.status_code,
+                    "response": response.text or "No response",
+                }
```

Comment:
So the point of returning the response.json() directly is that we _expect_ the Jira API to be stable, and to always return a json object - or throw an error for us to debug. This is an internal method used to as a wrapper around the JIRA API, so we don't need to modify the return values at all - it might actually raise unrelated errors if there's a type mismatch.

Reply:
Yes you are totally right but the thing is that for my usecase Jira returns a status code and nothing else, no body to get decoded to python json object which throws up an error so what I am trying to do here is mimic the behavior of other apis and creating a custom JSON here.
Please suggest me if there is a better way to do it

Reply:
That's weird.. so the jira api returns non-json on some endpoints, and we don't know which beforehand.
Can you check the response header? If its still `application/json` then we can go ahead with this method

Reply:
The header is some text/html
![image](https://github.com/user-attachments/assets/9f507132-65c8-46aa-8bda-4f4d776744af)

No response

![image](https://github.com/user-attachments/assets/6771ef21-c39c-414b-9480-8c6df6dbad1a)

Reply:
Okay, so I've just realized that 204 does in fact indicate that the response will be empty, so we can simply separate the condition for 204 into a secondary branch e.g. 
```python
if response.status_code == 204:
. . .
```